### PR TITLE
Tune the checkgroup config

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Minor tweaks to the checkgroup config.

### Does your PR introduce any breaking changes? If yes, please list them.

None

### CODE REVIEW

File: .github/checkgroup.yml [hunk 1]
```diff
@@ -163,8 +165,25 @@ subprojects:
...
+      - "pl-cpu (macOS-11, 3.7, oldest, stable)"
+      - "pl-cpu (ubuntu-20.04, 3.10, latest, stable)"
+      - "pl-cpu (ubuntu-20.04, 3.7, latest, stable)"
+      - "pl-cpu (ubuntu-20.04, 3.7, oldest, stable)"
+      - "pl-cpu (windows-2022, 3.10, latest, stable)"
+      - "pl-cpu (windows-2022, 3.7, latest, stable)"
+      - "pl-cpu (windows-2022, 3.7, oldest, stable)"
+      - "make-doctest (pytorch)"
+      - "make-html (pytorch)"
+      - "pytorch-lightning (GPUs)"
```
Comment (carmocca) [inline .github/checkgroup.yml:182]: Lite does not support HPU and IPU yet, so I haven't added their checks as required for Lite changes

Please confirm if that's okay @awaelchli. Maybe you want them to still run in each Lite PR for sanity

Comment (awaelchli) [reply .github/checkgroup.yml:182]: Hmm, we have utilities in Lite that are used by HPU and or IPU on the PL side. If this code would change in a bad way, we wouldn't see these checks fail on the PR, right? The risk is small, so I don't mind. On the other hand, IPU jobs are quite fast, so adding them wouldn't hurt I guess (except when the queue gets long)?

Comment (carmocca) [reply .github/checkgroup.yml:182]: The risk is small. If it was up to me, I would not add them until they break and we regret not having added them :stuck_out_tongue: 

I'll leave the decision up to you. Not adding them will mean faster development as you'd have less CI overhead.

**Review by awaelchli — APPROVED** [review]

File: .github/checkgroup.yml [hunk 2]
```diff
@@ -31,6 +31,7 @@ subprojects:
       - "setup.cfg"  # includes pytest config
       - ".github/workflows/docs-*.yml"
     checks:
+      # Note: updates here should be applied ot the lightning_lite group
```
Comment (otaj) [inline .github/checkgroup.yml:34]:
```suggestion
      # Note: updates here should be applied to the lightning_lite group
```

**Review by otaj — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Pull request merged**